src/EventStore.php

```diff
@@ -264,26 +250,28 @@ private function checkConnection(): void
         try {
             $request = new Request('GET', $this->url);
             $this->sendRequest($request);
-        } catch (RequestException $e) {
+        } catch (\GuzzleHttp\Exception\ConnectException $e) {
             throw new ConnectionFailedException($e->getMessage());
         }
     }
 
-    /**
-     * @param string $streamName
-     *
-     * @return string
-     */
     private function getStreamUrl(string $streamName): string
     {
         return sprintf('%s/streams/%s', $this->url, $streamName);
     }
 
+    private function removeCredentialsFromUrl(string $url): string
+    {
+        $parts = parse_url($url);
```

Comment:
this looks like a smell to me, do we need explicit credential handling in the EventStore? (see my comment in EventStoreTest)